### PR TITLE
Add extension trait for getting futures from AsyncDatagram

### PR DESCRIPTION
**Pull request by Nemo157**
**Choose one:** is this a 🐛 bug fix, a 🙋 feature, or a 🔦 documentation change?

## Checklist

- [ ] tests pass
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added

### CODE REVIEW

File: src/ext.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+use crate::AsyncDatagram;
+
+/// An extension trait that adds utility methods to [`AsyncDatagram`] types.
+pub trait AsyncDatagramExt: AsyncDatagram {
+  /// Creates a future that will send a given message to the specified target.
+  ///
+  /// The returned future will resolve to the number of bytes actually sent once the operation
+  /// completes.
+  fn send_to<'a>(
+    self: &'a mut Self,
```
Comment (taiki-e) [inline src/ext.rs]:
```suggestion
    &'a mut self,
```

File: src/ext.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      buf,
+      receiver,
+    }
+  }
+
+  /// Creates a future that will receive a message into the provided buffer.
+  ///
+  /// The returned future will resolve to the number of bytes received and target from whence the
+  /// data came  once the operation completes.
+  fn recv_from<'a>(self: &'a mut Self, buf: &'a mut [u8]) -> RecvFrom<'a, Self>
```
Comment (taiki-e) [inline src/ext.rs]:
```suggestion
  fn recv_from<'a>(&'a mut self, buf: &'a mut [u8]) -> RecvFrom<'a, Self>
```

**Review by taiki-e — APPROVED** [review]
LGTM

File: src/ext.rs (hunk 1 above)
Comment (Nemo157) [reply src/ext.rs]: Whoops, forgot I could simplify this after removing a `Pin`.

**Pull request merged**